File: python/1318_Minimum_Flips_to_Make_a_OR_b_Equal_to_c.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        ans = 0
        while a or b or c:
            if c & 1:
                ans += 0 if ((a & 1) or (b & 1)) else 1
            else:
                ans += (a & 1) + (b & 1)
            a >>= 1
            b >>= 1
            c >>= 1
        return ans

File: java/1318_Minimum_Flips_to_Make_a_OR_b_Equal_to_c.java
class Solution {
    public int minFlips(int a, int b, int c) {
        int ans = 0;
        while (a != 0 || b != 0 || c != 0) {
            if ((c & 1) == 1) {
                ans += ((a & 1) == 1 || (b & 1) == 1) ? 0 : 1; 
            } else {
                ans += (a & 1) + (b & 1);
            }
            a >>= 1;
            b >>= 1;
            c >>= 1;
        }
        return ans;
    }
}
